src/ParentNode.js

```diff
@@ -308,6 +338,7 @@ export default class ParentNode extends Node {
      */
     set innerHTML(html) {
         this._childNodes = [];
+        this._updateCollections();
```

Comment:
Nope, parse itself calls appendChild, etc. and structure updates automatically. I think we should clear all collections before parse to unlink elements that left in collections but already removed from DOM tree.